### DISCUSSION TITLE
Log rotation and retention

### DISCUSSION THREAD

**Discussion by pglavica** [keycloak/keycloak#10652] [Archive]
In KeyCloak distribution with Quarkus there is no explanation how log rotation and retention is handled.
Here is the official documentation: https://www.keycloak.org/server/logging

Apart from configuring log format and level, is there any options which define log rotation and retention policy or it is intended to be on OS-level (logadm, crontab...)

Regards,
Patrik

**Comment by rmjstn** [comment]
As described [here](https://www.keycloak.org/server/configuration#_unsupported_server_options), you can set a Quarkus log configuration in the file `conf/quarkus.properties`.

For example, in `conf/keycloak.conf` : 
```
log=console,file
log-file=/var/log/keycloak/keycloak.log
log-level=INFO
```

In `conf/quarkus.properties` : 
```
quarkus.log.file.rotation.max-file-size=1M
quarkus.log.file.rotation.max-backup-index=100
quarkus.log.file.rotation.rotate-on-boot=true
```

Rémy

**Reply by m4rc77** [reply]
See also https://github.com/keycloak/keycloak/issues/10618#issuecomment-1144827773 for details how it can be done
